crates/cli/src/utils/cmd.rs

```diff
@@ -383,10 +383,25 @@ pub async fn handle_traces(
     config: &Config,
     chain: Option<Chain>,
     labels: Vec<String>,
+    with_local_artifacts: bool,
     debug: bool,
     decode_internal: bool,
-    verbose: bool,
 ) -> Result<()> {
+    let (known_contracts, local_sources) = if with_local_artifacts {
+        let _ = sh_println!("Compiling project to generate artifacts");
+        let project = config.project()?;
+        let compiler = ProjectCompiler::new().quiet(true);
```

Comment:
```suggestion
        let compiler = ProjectCompiler::new();
```

given that uer explicitly asks for that it's probably not really useful and might be confusing if project takes long time to compile

Reply:
makes sense, I included change in https://github.com/foundry-rs/foundry/pull/7359/commits/6cec5eb169d3a0f5224d4a984193f9f14dc6ae80 together with test update